src/DataStructures/Tensor/EagerMath/OrthonormalOneform.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+/// \cond
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ *
+ * \brief Compute a spatial one-form orthonormal to the given unit form
+ *
+ * Given a unit spatial one-form \f$s_i\f$, compute a new form \f$t_i\f$
+ * which is orthonormal to \f$s_i\f$, in the sense that
+ * \f$\gamma^{ij}s_i t_j = 0\f$, for the given inverse spatial metric
+ * \f$\gamma^{ij}\f$.
+ *
+ * \details The new form is obtained via Gram-Schmidt process, starting
+ * from a form whose components are \f$t_i = \delta_i^I\f$, where \f$I\f$ is
+ * the index of the smallest absolute value of the components of \f$s_i\f$.
+ */
+template <typename DataType, size_t VolumeDim, typename Frame>
+void orthonormal_oneform(
+    gsl::not_null<tnsr::i<DataType, VolumeDim, Frame>*> orthonormal_form,
+    const tnsr::i<DataType, VolumeDim, Frame>& unit_form,
+    const tnsr::II<DataType, VolumeDim, Frame>& inv_spatial_metric) noexcept;
+
+template <typename DataType, size_t VolumeDim, typename Frame>
+tnsr::i<DataType, VolumeDim, Frame> orthonormal_oneform(
+    const tnsr::i<DataType, VolumeDim, Frame>& unit_form,
+    const tnsr::II<DataType, VolumeDim, Frame>& inv_spatial_metric) noexcept;
+// @}
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ *
+ * \brief Compute a spatial one-form orthonormal to the given unit forms
+ *
+ * Given a unit spatial one-form \f$s_i\f$ and another form \f$t_i\f$ that is
+ * orthonormal to \f$s_i\f$, compute a new form \f$u_i\f$ which is orthonormal
+ * to both \f$s_i\f$ and \f$t_i\f$, in the sense that
+ * \f$\gamma^{ij}s_i u_j = \f$\gamma^{ij}t_i u_j = 0\f$, for the given
+ * inverse spatial metric \f$\gamma^{ij}\f$.
+ *
+ * \details The new form is obtained by taking the covariant cross product
+ * of \f$s_i\f$ and \f$ t_i\f$. The spatial metric as well as is determinant
+ * must also be provided.
+ */
+template <typename DataType, size_t Dim, typename Frame>
```

Comment:
Thanks for noticing this. Yes, it is defined only for 3D, which is the only instance enabled in the code that generates the instantiations (cpp file), and so I should remove Dim here. The reason why I left it is because I was getting some compilation errors when testing the code for the char speeds for Valencia, which is generic in terms of Dim. I'll give it another try to see if I can get around those undefined references.